packages/components/src/disabled/index.js

```diff
@@ -94,10 +99,12 @@ class Disabled extends Component {
 			if ( focusable.hasAttribute( 'href' ) ) {
 				focusable.removeAttribute( 'href' );
 			}
-
-			// Disable all nodes from being tabbable order
-			focusable.setAttribute( 'tabindex', '-1' );
 		} );
+
+		// In FF the `<RichText>` node becomes focusable (presumanly via scripting).
+		// We manually disable this via tabindex also
+		const blockEditorEdgeCaseFocusable = '.block-editor-rich-text';
```

Comment:
@ellatrix might know better how this works. apparently `focus.focusable.find( this.node );` is not detecting these nodes as focusable.

Reply:
...and [here is a video](https://cld.wthms.co/ZzZKPW) of a `BlockPreview` component that is wrapped in a `Disabled` component. You can see that first

* For some reason `button` with `disabled` and `tabindex="=1"` is managing to be focusable (what!?)
* `<div class="editor-rich-text block-editor-rich-text" />` is focusable

The edge case above seems to fix both issues, but I have no idea why it becomes focusable in the first place.

Reply:
I have no idea. It should be regarded as focusable because it has `contenteditable="true"` as an attribute.

Reply:
I tried `wp.dom.focus.focusable.find( document.querySelector( '[data-type="core/paragraph"]' ) )` and it works fine.

Reply:
I _think_ I’ve figured it out…

As I suspected, it’s a FF specific thing. I cannot replicate in other browsers.

Basically, according to [the spec](https://html.spec.whatwg.org/multipage/interaction.html#focusable-area) an element which is a **scrollable container** is focusable in FF (a bit like as though it had tabindex="0". 

Also note this from the spec:

> The CSS ‘overflow’ property’s ‘scroll’ value typically creates a scrollable region.

Now [see my video of me updating the overflow to visible](https://cld.wthms.co/5a1HEE) on the div.block-editor-rich-text node. 

See how after I stop it being a scrollable container it no longer receives focus? Yep!

This explains why it isn't detected by our `focusable` utility and why my hack fix (in this PR) above does work.

So basically we need to 

1. Find a way to detect scrollable containers in our `focus` utility
2. Update RichText to not create a scrollable container

Reply:
Here's an idea for how we could detect these focusable scrollable containers:

```
const context = // some DOM node here

[...context.querySelectorAll('*')].filter(el => {
	const hasValidOverflow = getComputedStyle(el).overflow !== "visible"; // ouch this is expensive!
	const hasScroll = el.clientHeight >= el.scrollHeight; // and this :(

	return hasValidOverflow && hasScroll;
});
```

I'd suggest we'd need to make this an opt in feature as you'd need to be very careful about the perf impact of interrogating all those DOM nodes.

Reply:
> It is the wrapper element,

Which wrapping element? Is it part of `RichText`?

> Update RichText to not create a scrollable container

Where is it creating a scrollable container?

Reply:
> Which wrapping element? Is it part of RichText?

Yes. It's the one I linked to above which is part of `RichText`. Here's the full link to be completely unambiguous:

https://github.com/WordPress/gutenberg/blob/d63306d5e518486f0c074a9c5122d6b8cf7c83aa/packages/block-editor/src/components/rich-text/index.js#L417

> Where is it creating a scrollable container?

If you watch the video, you'll see me tabbing through the preview elements (these are all standard `button`s) without being dragged into the DOM of the preview itself because it has been disabled (via this PR). When I get to "Portfolio" however I end up with my tab focus being dragged onto the element referenced above. I then inspect it in Firefox devtools and you will see it is marked as `scroll`, which is Firefox's way of telling you that it is a "scrollable container".

I've since worked out that the scrollable container it is infact being created by styles being applied by the `core/gallery` block. Specifically the `overflow-y: auto`. 

https://github.com/WordPress/gutenberg/blob/2f9d0e20775083b3f1f99da2bc0f9e55ae78e6e0/packages/block-library/src/gallery/editor.scss#L35

If this is changed to `overflow: hidden` then the focus bug is completed removed.

Reply:
Ok @youknowriad @ellatrix, I've updated `focusable.find()` to have an option to detect scrollable containers and then applied this on the `Disable` component to ensure we truely disable all focus/tab stops within the wrapped components.

This fixes the bug I was experiencing.